src/Compilers/CSharp/Test/Semantic/Semantics/UnsafeTests.cs

```diff
@@ -7709,30 +7709,30 @@ static void Main()
                 // (6,34): error CS1586: Array creation must have array size or array initializer
                 //         { int* p = stackalloc int[]; }
                 Diagnostic(ErrorCode.ERR_MissingArraySize, "[]").WithLocation(6, 34),
-                // (8,34): error CS1586: Array creation must have array size or array initializer
-                //         { int* p = stackalloc int[][]; }
-                Diagnostic(ErrorCode.ERR_MissingArraySize, "[]").WithLocation(8, 34),
-                // (9,34): error CS1586: Array creation must have array size or array initializer
-                //         { int* p = stackalloc int[][1]; }
-                Diagnostic(ErrorCode.ERR_MissingArraySize, "[]").WithLocation(9, 34),
-                // (9,37): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
-                //         { int* p = stackalloc int[][1]; }
-                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "1").WithLocation(9, 37),
-                // (11,38): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
-                //         { int* p = stackalloc int[1][1]; }
-                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "1").WithLocation(11, 38),
                 // (7,31): error CS1575: A stackalloc expression requires [] after type
                 //         { int* p = stackalloc int[1, 1]; }
                 Diagnostic(ErrorCode.ERR_BadStackAllocExpr, "int[1, 1]").WithLocation(7, 31),
+                // (8,31): error CS0208: Cannot take the address of, get the size of, or declare a pointer to a managed type ('int[]')
+                //         { int* p = stackalloc int[][]; }
+                Diagnostic(ErrorCode.ERR_ManagedAddr, "int").WithArguments("int[]").WithLocation(8, 31),
```

Comment:
>ERR_ManagedAddr [](start = 37, length = 15)

The original diagnostic (about missing array size or initializer) was more informative.

Reply:
Isn't the real problem that a jagged array is a managed type? And thus you can't stackalloc it? That seems like what this error message is saying and that seems correct...